src/fs/hda_shares.go

```diff
@@ -194,3 +195,35 @@ func (shares *HdaShares) startMetadataPrefill(library *metadata.Library) {
 		}
 	}
 }
+
+func (shares *HdaShares) createThumbnailCache() {
+	time.Sleep(2 * time.Second)
+	go func() {
+		for {
+			select {
+			// watch for events
+			case event := <-watcher.Events:
+				op := event.Op.String()
+				log("EVENT: ", op)
+				switch {
+				case op == "CREATE" || op == "WRITE":
```

Comment:
In case of DELETE / RENAME too, we should have a case and handle it. For delete, remove the thumbnail if exists from the cache. Similarly for rename, rename its cached thumbnail too.